sg/initfile.go

```diff
@@ -49,6 +49,20 @@ func generateInitFile(g *codegen.File, pkg *doc.Package, mks []Makefile) error {
 		}
 		g.P("logger := ", g.Import("go.einride.tech/sage/sg"), ".NewLogger(\"", loggerName, "\")")
 		g.P("ctx = ", g.Import("go.einride.tech/sage/sg"), ".WithLogger(ctx, logger)")
+		g.P("ctx, cancel := context.WithCancel(ctx)")
+		g.P("defer cancel()")
+
+		g.P("go func() {")
+		g.P("shutdownCh := make(chan ", g.Import("os"), ".Signal, 1)")
+		g.P(g.Import("os/signal"), ".Notify(shutdownCh, ",
+			g.Import("os"), ".Interrupt, ",
+			g.Import("syscall"), ".SIGKILL, ",
+			g.Import("syscall"), ".SIGTERM)",
+		)
+		g.P("sig := <-shutdownCh")
+		g.P("logger.Printf(\"received: %s\", sig)")
+		g.P("cancel()")
```

Comment:
If we are already deferring the cancel call, do we need to call it here as well?

Reply:
because we want to cancel context in two cases:
1. if command finished on it's own. this is done by defering context
2. if we see an os signal. this line handles that. once context is cancelled, it will trigger a shutdown that will trigger all cmd.Cancel functions and kill every child

Reply:
Technically this would mean that the context is double cancelled if a Signal terminates us. This is not great, but also has no terrible sideeffects that I've been able to find or discern from a brief test like:
```
func main() {
	_, cancel := context.WithCancel(context.Background())
	cancel()
	cancel()
}
```

If you want to be super correct, observing the context and see whether it needs to be cancelled might be the "best", but seems like a lot of extra work for no benefit as my google-foo can find.

Reply:
Also, by checking out the code for how a cancel actually works, it seems indeed to be a safe operation to double cancel. https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/context/context.go;l=544-547
```
func (c *cancelCtx) cancel(removeFromParent bool, err, cause error) {
	if err == nil {
		panic("context: internal error: missing cancel error")
	}
	if cause == nil {
		cause = err
	}
	c.mu.Lock()
	if c.err != nil {
		c.mu.Unlock()
		return // already canceled
	}
```

Reply:
Agreed.

Reply:
I think it's ok. Cancelling context closes channel, so double cancelling should be fine 🤔

Reply:
however i agree, it is weird. i'll refactor this

Reply:
Imho, if you find a clean way to check it, do it. Otherwise just ignore it and move on - its a very minor thing and has no sideefects unless they break the Cancel functionality in a future Golang release.

Reply:
yea, i'd be surprised if they break it. this would mean closing the same channel twice 🙃

Reply:
didn't find a good way of refactoring that wouldn't introduce more complexity 🤔